packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/events/TouchEventDispatchTest.kt

```diff
@@ -0,0 +1,681 @@
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 2f, 1f, GESTURE_START_TIME, 1)
+        ),
+        listOf<WritableMap?>(
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 1f, 2f, GESTURE_START_TIME, 0),
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 2f, 1f, GESTURE_START_TIME, 1)
+        )
+      ),
+      /*
+       * CANCEL event:
+       * {
+       *   touch: 0,
+       *   touches: [],
+       *   changed: [touch0, touch1]
+       * }
+       * {
+       *   touch: 1,
+       *   touches: [],
+       *   changed: [touch0, touch1]
+       * }
+       */
+      buildGestureEvent(
+        SURFACE_ID,
+        TARGET_VIEW_ID,
+        1f,
+        3f,
+        GESTURE_START_TIME,
+        0,
+        emptyList<WritableMap>(),
+        listOf<WritableMap?>(
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 1f, 3f, GESTURE_START_TIME, 0),
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 2f, 1f, GESTURE_START_TIME, 1)
+        )
+      ),
+      buildGestureEvent(
+        SURFACE_ID,
+        TARGET_VIEW_ID,
+        2f,
+        1f,
+        GESTURE_START_TIME,
+        1,
+        emptyList<WritableMap>(),
+        listOf<WritableMap?>(
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 1f, 3f, GESTURE_START_TIME, 0),
+          buildGesture(SURFACE_ID, TARGET_VIEW_ID, 2f, 1f, GESTURE_START_TIME, 1)
+        )
+      )
+    )
+  private var dispatchedEvents: List<ReadableMap> = emptyList()
+  private lateinit var eventEmitter: FabricEventEmitter
+  private lateinit var uiManager: FabricUIManager
+
+  @Before
+  fun setUp() {
+    mockStatic(Arguments::class.java)
+    mockStatic(FabricUIManager::class.java)
+    whenever(Arguments.createArray()).thenAnswer { JavaOnlyArray() }
+    whenever(Arguments.createMap()).thenAnswer { JavaOnlyMap() }
+    val metrics = DisplayMetrics()
+    metrics.xdpi = 1f
+    metrics.ydpi = 1f
+    metrics.density = 1f
+    DisplayMetricsHolder.setWindowDisplayMetrics(metrics)
+    uiManager = mock(FabricUIManager::class.java)
+    eventEmitter = FabricEventEmitter(uiManager)
+  }
+
+  @Test
+  fun testFabric_startMoveEnd() {
+    for (event in startMoveEndSequence) {
+      event.dispatchModern(eventEmitter)
+    }
+    val argument = ArgumentCaptor.forClass(WritableMap::class.java)
+    verify(uiManager, times(4))
+      .receiveEvent(
+        ArgumentMatchers.anyInt(),
+        ArgumentMatchers.anyInt(),
+        ArgumentMatchers.anyString(),
+        ArgumentMatchers.anyBoolean(),
+        ArgumentMatchers.anyInt(),
+        argument.capture(),
+        ArgumentMatchers.anyInt()
+      )
+    Assert.assertEquals(startMoveEndExpectedSequence, argument.allValues)
+  }
+
+  @Test
+  fun testFabric_startMoveCancel() {
+    for (event in startMoveCancelSequence) {
+      event.dispatchModern(eventEmitter)
+    }
+    val argument = ArgumentCaptor.forClass(WritableMap::class.java)
+    verify(uiManager, times(6))
+      .receiveEvent(
+        ArgumentMatchers.anyInt(),
```

Comment:
Let's add an `import org.mockito.ArgumentMatchers.*` so we can just use `anyInt()` here and in all the various `ArgumentMatchers.*` place